gatorgrouper_cli.py

```diff
@@ -22,7 +22,13 @@
     if GG_ARGUMENTS.preferences is None:
         preference = None
     else:
-        preference = dict(read_student_file.read_csv_data(GG_ARGUMENTS.preferences))
+        preference = read_student_file.read_csv_data(GG_ARGUMENTS.preferences)
+        preference = [
+            [entry for entry in row if entry != ""]
+            for row in preference
+            if len(row) > 1
+        ]
+        preference = {row[0]: set(row[1:]) for row in preference}
```

Comment:
You should use a temporary variable for the intermediate decoding steps here; using `preference` could cause problems in the future, and confuses what the final value should be.

Reply:
Rather than use a temp variable I changed the `preferences` within the list comprehension to call `read_csv_data` directly. Does that help readability (or make it worse)?

Reply:
It actually makes it worse -- although it may make the code longer or less performant, two different names like `csv_data` or `validated_data` will make it much clearer. `preference` should only be set once.

Reply:
The latest commit distinguishes between the list `preference_list` and dict `preference`, setting `preference` only once -- is that in line with your suggestion or am I misunderstanding?

Reply:
Yes, perfect!